django/forms/fields.py

```diff
@@ -204,6 +422,18 @@ def has_changed(self, initial, data):
         data_value = data if data is not None else ''
         return initial_value != data_value
 
+    def bind_to_form(self, form, name):
+        """
+        Bind this field to a form with given name.
+        Returns an object that will eventually be used by a template when accessing
+        form fields.
+
+        For most fields this will just return a BoundField instance but can be
+        customized to return objects that behave differently or offer additional
+        functionality.
+        """
+        return BoundField(form, self, name)
```

Comment:
It would nice to have some example to justify the "more flexible" approach. Alternatively, instead of having a method to return a class, it seems sufficient to have an attribute on the form Field: `bound_field = BoundField`